lib/text_parser.dart

```diff
@@ -217,12 +218,23 @@ class TextParser extends StatelessWidget {
             optimizedChildren.add(child);
           }
         }
-        textSpan.children!.clear();
-        textSpan.children!.addAll(optimizedChildren);
+        textSpanChildren.clear();
+        textSpanChildren.addAll(optimizedChildren);
       }
     }
     // we don't care much about additional optimization for now
-    return textSpan;
+    return TextSpan(
```

Comment:
As discussed in chat, the `textSpan` might be a `LinkTextSpan`, and we should test for that. Something like this should do:

```dart
textSpan?.recognizer.dispose()
if (textSpan is LinkTextSpan) {
  return LinkTextSpan(......); // re-build it
} else {
  return TextSpan(......); // re-build it
}
```

Reply:
actually nevermind this, as optimisation is the last step what you have there should work